### DISCUSSION TITLE
PDF file to HL7

### DISCUSSION THREAD

**Discussion by Vnetherw** [nextgenhealthcare/connect#4935] [Q&A]
Hi
I use Mirth connect 3.6.1
I am working on a channel to read in a PDF file and output an HL7 message with the encapsulated PDF in OBX.5.5
The problem is that I keep getting a serialization error
ERROR MESSAGE: Error converting XML to ER7
org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.

Data type inbound is raw (filereader is set to read binary data type)
Data type outbound is HL7 v2.
HL7 message is built in the javascript transformer and appears as I expect.
MSH|||WHNT_DOC_PDF|WHNT|MMF|POSTIMAGE
EVN|
PID|||5639898||ZZZERS^CAROL
PV1|
OBR|||908ec38a-8021-4d78-9b82-53720faf5dbc|ASILetterPDF^ASI list validation letter||||||||||||||||||||MDOC|F
OBX|1||ASILetterPDF^ASI list validation letter^^^908ec38a-8021-4d78-9b82-53720faf5dbc.pdf||^APPLICATION^^BASE64^SlZCRVJpMHhMalF....z09DQo=||||||||ATTACHMENT

Earlier posts suggest this is due to the data type not matching the message but not sure what I should select instead of raw.
Have tried moving the transformer from source to destination but this did not make a difference.
Any suggestions would be welcomed.

**Comment by pacmano1** [comment]
RAW is correct for the inbound type.    

Export your channel and post it here.

**Comment by Vnetherw** [comment]
I found that changing the outbound message type to raw prevented this error.
The message is successfully passed on to another channel which is set to receive HL7v2

**Comment by pacmano1** [comment]
If you are building an HL7 message with the PDF encapsulated in OBX-5.5, you should not have to set the outbound message type to raw so something else is incorrect in your channel.

**Comment by tonygermano** [comment]
@Vnetherw Are you building the hl7 as xml or as a string? If you are building it as a string, then Raw is the correct outbound type to use, though this is not the most common way to do it.
